src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp

```diff
@@ -530,7 +530,6 @@ class ShenandoahHeap : public CollectedHeap {
   }
 
   ShenandoahOldGeneration*   old_generation()    const {
-    assert(mode()->is_generational(), "Old generation requires generational mode");
```

Comment:
Can we assert that `ShenandoahCardBarrier` is on instead of removing this assertion?

Reply:
Sure